src/app/asset-grid/asset-grid.component.ts

```diff
@@ -655,6 +656,7 @@ export class AssetGrid implements OnInit, OnDestroy {
           this.allResults = allThumbnails
           this.results = this.allResults.slice(0)
           this.visibleResults = this.results.filter(asset => asset.status !== 'not-available')
+          this.invisibleResults = this.results.filter(asset => asset.status === 'not-available')
```

Comment:
I suggested `filter` when we just had one array, but now that we have two arrays it might make more sense to iterate over it just once. We could still use a `forEach` to stay with the control flow statements and inside the method conditionally push onto the visible/invisible results array.

Reply:
Done!